core/src/main/java/apoc/trigger/TriggerHandlerWrite.java

```diff
@@ -0,0 +1,153 @@
+package apoc.trigger;
+
+import apoc.SystemLabels;
+import apoc.SystemPropertyKeys;
+import apoc.util.Util;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.internal.helpers.collection.Pair;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static apoc.ApocConfig.APOC_TRIGGER_ENABLED;
+import static apoc.ApocConfig.apocConfig;
+
+public class TriggerHandlerWrite {
+    public static final String NOT_ENABLED_ERROR = "Triggers have not been enabled." +
+            " Set 'apoc.trigger.enabled=true' in your apoc.conf file located in the $NEO4J_HOME/conf/ directory.";
+
+    public static Map<String, Object> toTriggerInfo(Node node) {
+        return node.getAllProperties()
+                .entrySet().stream()
+                .filter(e -> !List.of(SystemPropertyKeys.name.name(), SystemPropertyKeys.database.name()).contains(e.getKey()))
+                .collect(HashMap::new, // workaround for https://bugs.openjdk.java.net/browse/JDK-8148463
+                        (mapAccumulator, e) -> {
+                            Object value = List.of(SystemPropertyKeys.selector.name(), SystemPropertyKeys.params.name()).contains(e.getKey()) 
+                                    ? Util.fromJson((String) e.getValue(), Map.class) 
+                                    : e.getValue();
+                            
+                            mapAccumulator.put(e.getKey(), value);
+                    }, HashMap::putAll);
+    }
+
+    private static boolean isEnabled() {
+        return apocConfig().getBoolean(APOC_TRIGGER_ENABLED);
+    }
+
+    public static void checkEnabled() {
+        if (!isEnabled()) {
+            throw new RuntimeException(NOT_ENABLED_ERROR);
+        }
+    }
+
+    public static Map<String, Object> install(String databaseName, String triggerName, String statement, Map<String,Object> selector, Map<String,Object> params) {
+        checkEnabled();
+        final HashMap<String, Object> previous = new HashMap<>();
+
+        withSystemDb(tx -> {
+            Node node = Util.mergeNode(tx, SystemLabels.ApocTrigger, null,
```

Comment:
It seems like both the new Triggers functionality and the deprecated Triggers functionality is using the same node label.

I'm not sure that we should really be consider the new triggers equivalent to the old triggers. For example, should the new apoc.triggers.stop procedure be able to stop the old triggers? Should the apoc.trigger.add (deprecated) be able to add the new triggers? 

It doesn't seem to me like these new procedures need to be backwards-compatible with the old ones, and so I'd probably rather keep them separate entirely. I imagine that it will help us and our users debug support issues when users are migrating from the old procedures to the new ones, as will probably help us avoid a great deal of confusion. 

But I'm happy to hear counter-arguments, of course!

Reply:
For me the argument for having the same levels is you want people with triggers in their database to still work. We are recommending to use the new procedures, the old ones will be deprecated and deleted. Unless we provide a migration path, we cannot let people that have triggers installed already without support in a couple years.

Reply:
This was fixed :+1: